src/modules/tests/tests.reducer.ts

```diff
@@ -43,17 +32,37 @@ const initialState: TestsModel = {
  */
 export function testsReducer(
   state = initialState,
-  action: Action,
-) {
-  if (action.type === LOAD_PERSISTED_TESTS_SUCCESS) {
-    return (<LoadPersistedTestsSuccess>action).tests;
-  }
+  action: testActions.Types | journalActions.JournalActionTypes) {
 
   const slotId = deriveSlotId(state, action);
-  if (!slotId) {
-    return state;
+
+  switch (action.type){
+    case testActions.LOAD_PERSISTED_TESTS_SUCCESS:
+      return (<testActions.LoadPersistedTestsSuccess>action).tests;
+    case testActions.SET_ACTIVITY_CODE:
+      return {
+        ...state,
+        startedTests: {
+          ...state.startedTests,
+          [slotId]: {
+            ...state.startedTests[slotId],
+            activityCode: action.payload,
+          },
+        },
+      };
+    default:
+      return slotId ? createStateObject(state, action, slotId) : state;
+  }
+}
+
+const deriveSlotId = (state: TestsModel, action: Action): string | null => {
+  if (action instanceof journalActions.StartTest) {
```

Comment:
Nice refactor of this, needs a rebase to pick up the additional action `journalActions.ActivateTest` that will resolve a slot ID.

Reply:
Just tried to rebase and there is nothing to add to this pr as I'm on the latest code

Reply:
I'm referring to this line:
https://github.com/dvsa/mes-mobile-app/blob/develop/src/modules/tests/tests.reducer.ts#L98

Reply:
Thanks had missed that in the rebase, just pushing up a fix for it :)